async_lru/__init__.py

```diff
@@ -222,6 +222,7 @@ async def __call__(self, /, *fn_args: Any, **fn_kwargs: Any) -> _R:
         if self.__maxsize is not None and len(self.__cache) > self.__maxsize:
             dropped_key, cache_item = self.__cache.popitem(last=False)
             cache_item.cancel()
+            del cache_item
```

Comment:
I'm not clear how this relates to exceptions specifically, but here's my understanding of the issue, which should be documented:
```suggestion
            # While waiting for future below, we want dropped cache_item to be garbage
            # collected. Without the del, it is possible to end up with a chain of cache_item
            # references stopping them from being garbage collected at all.
            del cache_item
```

Reply:
Should we be doing dropped_key too?